### PR TITLE
Arc-2229 add analytics UI events

### PR DESCRIPTION
**Pull request by rachellerathbone**
Added UI events to most things a user can click, excluding form submit buttons (they fall under track events).

Small sampling of some of the UI events added:

https://github.com/atlassian/jenkins-for-jira/assets/37155488/526aeadf-e1fd-4455-be94-2406bf57b917

### CODE REVIEW

File: app/jenkins-for-jira-ui/src/components/JenkinsConfigurationForm/JenkinsConfigurationForm.tsx [hunk 1]
```diff
@@ -56,17 +62,38 @@ const JenkinsConfigurationForm = ({
 	setHasError,
 	isLoading
 }: JenkinsConfigurationFormProps) => {
+	const analyticsClient = new AnalyticsClient();
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/JenkinsConfigurationForm/JenkinsConfigurationForm.tsx:65]: Not sending analytics inside useEffect in this component so fine to define here. Thoughts on moving outside the component though just for consistencies sake?

File: app/jenkins-for-jira-ui/src/components/EmptyState/EmptyState.tsx [hunk 2]
```diff
@@ -21,10 +25,28 @@ const EmptyStateJenkins = () => {
 	}, []);
 
 	const onClickConnect = () => {
+		analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.UiEvent,
+			AnalyticsUiEventsEnum.ConnectJenkinsServerEmptyStateName,
+			{
+				source: AnalyticsScreenEventsEnum.ConfigurationEmptyStateScreenName,
+				action: 'clickedConnectJenkinsServerEmptyState',
```
Comment (krazziekay) [inline app/jenkins-for-jira-ui/src/components/EmptyState/EmptyState.tsx:33]: Can we keep the format for the action consistent? Like `clicked Connect Jenkins server empty state`.

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/ConnectedServers.tsx [hunk 3]
```diff
@@ -119,17 +123,48 @@ const ConnectedServersTable = ({ jenkinsServerList, refreshServers }: ConnectedS
...
 	}, [jenkinsServerList]);
 
-	const onClickManage = (jenkinsServerUuid: string) => {
+	const onClickManage = async (jenkinsServerUuid: string, serverName: string) => {
+		await analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.UiEvent,
+			AnalyticsUiEventsEnum.ManageConnectionConfiguredStateName,
+			{
+				source: AnalyticsScreenEventsEnum.ConfigurationConfiguredStateScreenName,
+				action: 'clickedManageConnection',
```
Comment (krazziekay) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/ConnectedServers.tsx:132]: Same here, `clicked manage connection`

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/ConnectedServers.tsx [hunk 4]
```diff
@@ -146,12 +181,32 @@ const ConnectedServersTable = ({ jenkinsServerList, refreshServers }: ConnectedS
...
+	const closeConfirmServerDisconnect = async () => {
 		setShowConfirmServerDisconnect(false);
 		setIsLoading(false);
+
+		await analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.UiEvent,
+			AnalyticsUiEventsEnum.DisconnectServerModalClosedConfiguredStateName,
+			{
+				source: AnalyticsScreenEventsEnum.ConfigurationConfiguredStateScreenName,
+				action: 'clicked cancel or disconnected server',
```
Comment (krazziekay) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/ConnectedServers.tsx:206]: Having `or` could be confusing when analyzing data. This method is triggered on the dismissal of the modal, so should be just `clicked cancel`.

Comment (krazziekay) [reply app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/ConnectedServers.tsx:206]: Then I think it should only be `dismissed the modal`. 
We already have events sent when the user cancels or disconnects right, then if we want to get both of the above scenarios, we first check if the user has clicked cancel or disconnect followed by this dismissal action.

**Review by krazziekay — APPROVED** [review]

**Pull request merged**